nodes.py

```diff
@@ -272,7 +258,59 @@ def nodefor(self, element):
             "ulink": UlinkNode,
             "verbatim": VerbatimNode,
             None: Node,
-        }[element.name]
+        }
+        if element.name == "compounddef":
+            if element["kind"] == "page":
+                return defaults[element.name]
+            else:
+                return {
+                    "group": GroupNode,
+                }[element["kind"]]
+
+        if element.name == "sectiondef":
+            return {
+                "define": DefineSectiondefNode,
+                "enum": EnumSectiondefNode,
+                "typedef": TypedefSectiondefNode,
+                "func": FunctionSectiondefNode,
+                "var": VariableSectiondefNode,
+                "user-defined": UserDefinedSectiondefNode,
+            }[element["kind"]]
+
+        if element.name == "memberdef":
+            return {
+                "define": DefineMemberdefNode,
+                "enum": EnumMemberdefNode,
+                "typedef": TypedefMemberdefNode,
+                "function": FunctionMemberdefNode,
+                "variable": VariableMemberdefNode,
+            }[element["kind"]]
+
+        if element.name == "title":
+            if self.has_parent(element, "compounddef", ["kind", "page"]):
+                return {
+                    "title": HeadingNode
+                }[element.name]
+            else:
+                return defaults[element.name]
+
+        if element.name == "detaileddescription":
+            # because of the block context wrapping, we need to check 2 levels up
+            if self.has_parent(element, "compounddef", ["kind", "page"]):
+                return {
+                    "detaileddescription": SimplesectNode
+                }[element.name]
+            else:
+                return defaults[element.name]
+
+        if element.name == "compoundname":
+            # because of the block context wrapping, we need to check 2 levels up
+            if self.has_parent(element, "compounddef", ["kind", "page"]):
+                return NoneNode
+            else:
+                return defaults[element.name]
```

Comment:
I guess these can't all be combined into the following because you need to refer to the `default` mapping for all other element names?

```suggestion
        if self.has_parent(element, "compounddef", ["kind", "page"]):
            return {
                "title": HeadingNode,
                "detaileddescription": SimplesectNode,
                "compoundname": NoneNode,
            }[element.name]
```

Reply:
I was just sticking with your architecture here, but happy to simplify!

Reply:
I was just sticking with your architecture here, but happy to simplify! We *do* need to filter out all other elements somehow, but I just pushed a somewhat simplified version:

```
special_cases = ["title", "detaileddescription", "compoundname"]
if element.name in special_cases and self.has_parent(element, "compounddef", ["kind", "page"]):
    return {
        "title": HeadingNode,
        "detaileddescription": SimplesectNode,
        "compoundname": NoneNode,
    }[element.name]
```